sos/policies/distros/redhat.py

```diff
@@ -430,12 +432,30 @@ def upload_sftp(self):
                                                        password=_token)
         raise Exception("Could not retrieve valid or anonymous credentials")
 
+    def check_file_too_big(self, archive):
+        size = os.path.getsize(archive)
+        # Lets check if the size is bigger than the limit.
+        # There's really no need to transform the size to Gb,
+        # so we don't need to call any size converter implemented
+        # in tools.py
+        if (size >= self._max_size_request) and\
+                self.get_upload_url().startswith(RH_API_HOST):
```

Comment:
The check against the upload URL should be done prior to the call to `check_file_too_big()`, otherwise we are doing the size comparison needlessly.